util/canonical.go

```diff
@@ -89,7 +90,9 @@ func ConnectionFromCanonical(s string) *Connection {
 		return nil
 	}
 
-	ip := net.ParseIP(s[0 : i-1])
+	bracketTruncatedIP := truncateIPv6Bracket(s[0 : i-1])
```

Comment:
I removed this method because it seems unnecessary.

Reply:
ok